File: app/src/main/java/com/textfield/json/outabus/GenericActivity.java
package com.textfield.json.outabus;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.widget.ListView;
import android.widget.TextView;

public class GenericActivity extends AppCompatActivity {
    protected ListView listView;
    protected Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.generic_layout);
        toolbar = (Toolbar) findViewById(R.id.generic_toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeButtonEnabled(true);
        listView = (ListView) findViewById(R.id.generic_listview);
    }

    public void showProgressBar(boolean show) {

    }

    public void setSmallTitle(String text) {
        toolbar.setTitle("");
        TextView title = (TextView) toolbar.findViewById(R.id.textView);
        title.setText(text);
        setSupportActionBar(toolbar);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            super.onBackPressed();
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/textfield/json/outabus/activities/GenericActivity.kt
package com.textfield.json.outabus.activities

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.MenuItem
import android.widget.ListView
import android.widget.TextView

import com.textfield.json.outabus.R

open class GenericActivity : AppCompatActivity() {
    protected lateinit var listView: ListView
    protected lateinit var toolbar: Toolbar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.generic_layout)
        toolbar = findViewById<Toolbar>(R.id.generic_toolbar)
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeButtonEnabled(true)
        listView = findViewById<ListView>(R.id.generic_listview)
    }

    fun showProgressBar(show: Boolean) {

    }

    fun setSmallTitle(text: String) {
        toolbar.title = ""
        val title = toolbar.findViewById<TextView>(R.id.textView)
        title.text = text
        setSupportActionBar(toolbar)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}
